constructor/shar.py

```diff
@@ -173,12 +173,21 @@ def create(info, verbose=False):
     pre_t.addfile(tarinfo=tarfile.TarInfo("conda-meta/history"))
     post_t.add(join(tmp_dir, "conda-meta", "history"), "conda-meta/history")
 
+    if info.get("freeze_base"):
+        post_t.add(join(tmp_dir, "conda-meta", "frozen"), "conda-meta/frozen")
+
     for env_name in info.get("_extra_envs_info", {}):
         pre_t.addfile(tarinfo=tarfile.TarInfo(f"envs/{env_name}/conda-meta/history"))
         post_t.add(
             join(tmp_dir, "envs", env_name, "conda-meta", "history"),
             f"envs/{env_name}/conda-meta/history",
         )
+        env_config = info["_extra_envs_info"][env_name]
```

Comment:
```suggestion
    for env_name, env_config in info.get("_extra_envs_info", {}).items():
        pre_t.addfile(tarinfo=tarfile.TarInfo(f"envs/{env_name}/conda-meta/history"))
        post_t.add(
            join(tmp_dir, "envs", env_name, "conda-meta", "history"),
            f"envs/{env_name}/conda-meta/history",
        )
```

If you use the value, you can just iterate over the items - it's a little cleaner.

Reply:
I thought about doing this too when I first decided to move my logic here.

Reply:
I ended up not doing this since we're checking for if the files exist instead of using the config logic